gk/main.c

```diff
@@ -2132,41 +2172,43 @@ add_ggu_policy_bulk(struct gk_add_policy **policies, int num_policies,
 
 	for (i = 0; i < num_policies; i++) {
 		int pos = pos_arr[i];
-		struct flow_entry *fe;
-		struct ggu_policy *policy = &policies[i]->policy;
 
-		if (pos >= 0)
-			fe = &instance->ip_flow_entry_table[pos];
-		else if (instance->num_scan_del > 0)
-			continue;
-		else {
-			struct gk_fib *fib = look_up_fib(
-				&gk_conf->lpm_tbl, &policy->flow);
-			if (fib == NULL || fib->action !=
-					GK_FWD_GRANTOR) {
-				/*
-				 * Drop this solicitation to add
-				 * a policy decision.
-				 */
-				char err_msg[128];
-				int ret = snprintf(err_msg, sizeof(err_msg),
-					"gk: at %s initialize flow entry error",
-					__func__);
-				RTE_VERIFY(ret > 0 &&
-					ret < (int)sizeof(err_msg));
-				print_flow_err_msg(
-					&policy->flow, err_msg);
-				continue;
-			}
+		if (pos >= 0) {
+			struct ggu_policy *policy =
+				&policies[i]->policy;
+			struct flow_entry *fe =
+				&instance->ip_flow_entry_table[pos];
 
-			fe = add_new_flow_from_policy(fib,
-				policy, instance, gk_conf,
-				flow_hash_val_arr[i]);
-			if (fe == NULL)
-				continue;
+			update_flow_entry(fe, policy);
+		} else if (instance->num_scan_del > 0)
+			continue;
```

Comment:
Let's move this test just after the `for` loop and do an early return from there.

Reply:
By after, I meant after the body of the `for` loop; before calling `lookup_fib_bulk()`.